src/WebJobs.Extensions/Extensions/Core/Monitoring/SlidingWindowTraceFilter.cs

```diff
@@ -56,11 +59,12 @@ public override string Message
         public int Threshold { get; private set; }
 
         /// <inheritdoc/>
-        public override Collection<TraceEvent> Events
+        public override IEnumerable<TraceEvent> GetEvents()
         {
-            get
+            // locking to prevent race condition while accessing events
+            lock (_eventsLock)
```

Comment:
Don't need the new _eventsLock - just use the existing _syncLock which is the existing guard used in other places on this collection.